src/app/clusters/boolean-state-configuration-server/boolean-state-configuration-server.cpp

```diff
@@ -0,0 +1,443 @@
+
+static_assert(kBooleanStateConfigurationDelegateTableSize <= kEmberInvalidEndpointIndex,
+              "BooleanStateConfiguration Delegate table size error");
+
+static CHIP_ERROR StoreCurrentSensitivityLevel(EndpointId ep, uint8_t level);
+
+namespace {
+Delegate * gDelegateTable[kBooleanStateConfigurationDelegateTableSize] = { nullptr };
+
+Delegate * GetDelegate(EndpointId endpoint)
+{
+    uint16_t ep = emberAfGetClusterServerEndpointIndex(endpoint, BooleanStateConfiguration::Id,
+                                                       EMBER_AF_BOOLEAN_STATE_CONFIGURATION_CLUSTER_SERVER_ENDPOINT_COUNT);
+    return (ep >= kBooleanStateConfigurationDelegateTableSize ? nullptr : gDelegateTable[ep]);
+}
+
+bool isDelegateNull(Delegate * delegate)
+{
+    if (delegate == nullptr)
+    {
+        return true;
+    }
+    return false;
+}
+
+class BooleanStateConfigAttrAccess : public AttributeAccessInterface
+{
+public:
+    BooleanStateConfigAttrAccess() : AttributeAccessInterface(Optional<EndpointId>::Missing(), BooleanStateConfiguration::Id) {}
+
+    CHIP_ERROR Write(const ConcreteDataAttributePath & aPath, AttributeValueDecoder & aDecoder) override;
+    CHIP_ERROR Read(const ConcreteReadAttributePath & aPath, AttributeValueEncoder & aEncoder) override;
+
+private:
+    CHIP_ERROR WriteCurrentSensitivityLevel(const ConcreteDataAttributePath & aPath, AttributeValueDecoder & aDecoder);
+    CHIP_ERROR ReadCurrentSensitivityLevel(const ConcreteReadAttributePath & aPath, AttributeValueEncoder & aEncoder);
+};
+
+BooleanStateConfigAttrAccess gAttrAccess;
+
+CHIP_ERROR BooleanStateConfigAttrAccess::WriteCurrentSensitivityLevel(const ConcreteDataAttributePath & aPath,
+                                                                      AttributeValueDecoder & aDecoder)
+{
+    uint8_t curSenLevel;
+    ReturnErrorOnFailure(aDecoder.Decode(curSenLevel));
+    // VerifyOrReturnError(EMBER_ZCL_STATUS_SUCCESS == SupportedSensitivityLevels::Get(aPath.mEndpointId, &supportedSensLevel),
+    //                     CHIP_IM_GLOBAL_STATUS(Failure));
+
+    // VerifyOrReturnError(curSenLevel < supportedSensLevel, CHIP_IM_GLOBAL_STATUS(ConstraintError));
+    // ReturnErrorOnFailure(GetSafeAttributePersistenceProvider()->WriteScalarValue(
+    //     ConcreteAttributePath(aPath.mEndpointId, aPath.mClusterId, CurrentSensitivityLevel::Id), curSenLevel));
+
+    return StoreCurrentSensitivityLevel(aPath.mEndpointId, curSenLevel);
+}
+
+CHIP_ERROR BooleanStateConfigAttrAccess::ReadCurrentSensitivityLevel(const ConcreteReadAttributePath & aPath,
+                                                                     AttributeValueEncoder & aEncoder)
+{
+    uint8_t senLevel;
+    CHIP_ERROR err = GetSafeAttributePersistenceProvider()->ReadScalarValue(
+        ConcreteAttributePath(aPath.mEndpointId, aPath.mClusterId, CurrentSensitivityLevel::Id), senLevel);
```

Comment:
Why is this constructing a new path instead of just passing aPath?

Reply:
My bad.